liquibase-standard/src/main/java/liquibase/analytics/UsageAnalyticsFactory.java

```diff
@@ -0,0 +1,27 @@
+package liquibase.analytics;
+
+import liquibase.Scope;
+import liquibase.plugin.AbstractPluginFactory;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class UsageAnalyticsFactory extends AbstractPluginFactory<UsageAnalyticsListener> {
+    @Override
+    protected Class<UsageAnalyticsListener> getPluginClass() {
+        return UsageAnalyticsListener.class;
+    }
+
+    @Override
+    protected int getPriority(UsageAnalyticsListener obj, Object... args) {
+        return obj.getPriority();
+    }
+
+    public void handleEvent(Event event) {
```

Comment:
Are we planing to add some threading around here to make it non-blocking?

Reply:
I am not sure how valuable the threading here would be. The handleEvent call gets called when the `CommandScope#execute` method exits, so it's essentially right before Liquibase shuts down. So it would be in a separate thread but nothing else would be executing at the same time.

My intention is to have a timeout, so it attempts to deliver the telemetry for a set amount of time before giving up.

Reply:
oh i see, so we are only sending the data after liquibase executes and not at startup. Alright then!